lms/djangoapps/labster_accounts/management/commands/obfuscate_user_data.py

```diff
@@ -0,0 +1,154 @@
+
+                    all_users = chain(not_enrolled_users, enrolled_users)
+
+                    user_emails |= set(all_users)
+                except CourseLicense.DoesNotExist:
+                    pass
+
+        if emails:
+            user_emails |= set(emails)
+
+        self.obfuscate_users(user_emails)
+
+    def obfuscate_users(self, user_emails):
+        """
+        Obfuscates user model fields: names, email.
+        """
+        users = User.objects.filter(email__in=user_emails)
+
+        field_actions = {
+            'first_name': self.generate_random_string,
+            'last_name': self.generate_random_string,
+            'username': self.generate_random_string,
+            'profile.name': self.generate_random_string,
+            'profile.mailing_address': self.generate_random_string,
+            'email': self.obfuscate_email
+        }
+
+        self.obfuscate_names_emails(users, field_actions)
+
+    @staticmethod
+    def obfuscate_names_emails(query_set, field_actions):
+        """
+        Obfuscates objects name, email and other fields.
+        """
+        fields_select = set([key.split('.')[0] if '.' in key else key for key in field_actions.keys()])
+        for obj in query_set.only(*fields_select):
+            print("Obfuscating data for %s" % obj)
+            values = {}
+            references = {}
+            for field, action in field_actions.items():
+                action = action_decorator(action, field)
+                if '.' in field:
+                    # one2one relation or foreign key found
+                    names = field.split('.')
+                    # get relation field value
+                    val = getattr(getattr(obj, names[0]), names[1], '')
+                    if not (names[0] in references):
+                        references[names[0]] = {}
+                    references[names[0]].update({names[1]: action(val)})
+                else:
+                    val = getattr(obj, field, '')
+                    values[field] = action(val)
+
+            if values:
+                obj.__class__.objects.filter(id=obj.id).update(**values)
+
+            if references:
+                for rel, item in references.items():
+                    _class = getattr(obj, rel).__class__
+                    _id = getattr(obj, rel).id
+                    _class.objects.filter(id=_id).update(**item)
+
+    @staticmethod
+    def generate_random_string(text):
+        """
+        Create a string of random characters of specified length.
+        """
+        chars = [char for char in string.ascii_lowercase]
+        return string.join((random.choice(chars) for __ in range(len(text))), '')
+
+    def obfuscate_email(self, email):
+        """
+        Return random string of the same length as input value.
+        """
+        obfuscated = self.generate_random_string(email)
+        return ''.join([obfuscated[idx] if ch not in ['@', '.'] else ch for idx, ch in enumerate(email)])
```

Comment:
Cannot we simply do `RANDOM_STRING@example.com?`

Reply:
good catch :), changed